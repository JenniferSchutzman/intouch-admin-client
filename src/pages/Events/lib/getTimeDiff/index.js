import moment from 'moment-timezone'

/**
 * getTimeDiff - if there is an end time, returns the difference between start and end time
 * @param {object} time
 */
const getTimeDiff = time => {
  const start = moment(time.string, 'h:mm a')
  const end = moment(time.endString, 'h:mm a')
  const diff = end.diff(start, 'minutes')
  const formatMins = diff => (diff % 60 === 0 ? null : diff % 60)
  const diffHour = Math.floor(diff / 60)
  const res =
    diff / 60 >= 1
      ? {
          hours: diffHour,
          minutes: formatMins(diff)
        }
      : {
          hours: 0,
          minutes: diff
        }

  return res
}

export default getTimeDiff
