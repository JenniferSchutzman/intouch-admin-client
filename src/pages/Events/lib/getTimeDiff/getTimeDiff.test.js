import getTimeDiff from './'

test('getTimeDiff should work with only minutes', () => {
  const result = getTimeDiff(onlyMinutesData)
  expect(result).toMatchSnapshot()
})
test('getTimeDiff should work with only hours', () => {
  const result = getTimeDiff(onlyHoursData)
  expect(result).toMatchSnapshot()
})
test('getTimeDiff should work with hours and minutes', () => {
  const result = getTimeDiff(hoursAndMinutesData)
  expect(result).toMatchSnapshot()  
})

const onlyMinutesData = {
  string: '12:00 pm',
  endString: '12:05 pm',
  hasEndTime: true,
  unix: 1524931200,
  endUnix: 1524931500
}

const onlyHoursData = {
  string: '4:00 pm',
  endString: '6:00 pm',
  hasEndTime: true,
  unix: 1524945600,
  endUnix: 1524952800
}

const hoursAndMinutesData = {
  string: '7:00 pm',
  endString: '9:06 pm',
  hasEndTime: true,
  unix: 1524956400,
  endUnix: 1524963960
}
