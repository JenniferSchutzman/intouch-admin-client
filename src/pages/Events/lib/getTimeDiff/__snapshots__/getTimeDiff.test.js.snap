// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getTimeDiff should work with hours and minutes 1`] = `
Object {
  "hours": 2,
  "minutes": 6,
}
`;

exports[`getTimeDiff should work with only hours 1`] = `
Object {
  "hours": 2,
  "minutes": null,
}
`;

exports[`getTimeDiff should work with only minutes 1`] = `
Object {
  "hours": 0,
  "minutes": 5,
}
`;
